server/pulp/server/managers/repo/unit_association.py

```diff
@@ -414,15 +414,21 @@ def calculate_associated_type_ids(source_repo_id, associated_units):
 
 
 def create_transfer_units(associate_units, associated_unit_type_ids):
-    type_defs = {}
-    for def_id in associated_unit_type_ids:
-        type_def = types_db.type_definition(def_id)
-        type_defs[def_id] = type_def
+    unit_key_fields = {}
+    for unit_type_id in associated_unit_type_ids:
+        type_def = types_db.type_definition(unit_type_id)
+        if type_def is not None:
```

Comment:
Consider simplifying the `if` statement to not have a negation. The new-style could become the `if` codepath with

```
if type_def is None:
```